File: C++/A/146A-luckyticket.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    int n;
    string ticket;
    cin >> n >> ticket;
    for (char c : ticket)
        if (c != '4' && c != '7')
        {
            cout << "NO";
            return 0;
        }
    int half = n / 2;
    string first_half = ticket.substr(0, half);
    string second_half = ticket.substr(half);
    int first_sum = 0, second_sum = 0;
    for (char c : first_half)
        first_sum += c - '0';
    for (char c : second_half)
        second_sum += c - '0';
    bool flag = first_sum == second_sum;
    cout << ((flag) ? "YES" : "NO");
}

File: Python/A/146A-luckyticket.py
n = int(input())
string = input().strip()
for char in string:
    if char != '4' and char != '7':
        print('NO')
        exit()

half = n // 2
first_half = string[:half]
second_half = string[half:]
first_sum = second_sum = 0
for char in first_half:
    first_sum += ord(char)
for char in second_half:
    second_sum += ord(char)
flag = first_sum == second_sum

print('YES' if flag else 'NO')
